homeassistant/components/rpi_gpio.py

```diff
@@ -5,64 +5,131 @@
 https://home-assistant.io/components/rpi_gpio/
 """
 import logging
+import importlib
+
+import voluptuous as vol
 
 from homeassistant.const import (
     EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
+import homeassistant.helpers.config_validation as cv
 
-REQUIREMENTS = ['RPi.GPIO==0.6.5']
+REQUIREMENTS = ['RPi.GPIO==0.6.5',
+                'OrangePi.GPIO==0.6.3']
 
 _LOGGER = logging.getLogger(__name__)
 
+CONF_BOARD_FAMILY = 'board_family'
+CONF_BOARD = 'board'
+
+DEFAULT_FAMILY = 'raspberry_pi'
+FAMILY_LIBRARIES = {'raspberry_pi': 'RPi.GPIO',
+                    'orange_pi': 'OPi.GPIO'}
+
+ORANGEPI_BOARDS = {'zero': 'ZERO',
+                   'r1': 'R1',
+                   'zeroplus': 'ZEROPLUS',
+                   'zeroplus2h5': 'ZEROPLUS2H5',
+                   'zeroplus2h3': 'ZEROPLUS2H3',
+                   'pcpplus': 'PCPCPLUS',
+                   'one': 'ONE',
+                   'lite': 'LITE',
+                   'plus2e': 'PLUS2E',
+                   'pc2': 'PC2',
+                   'prime': 'PRIME',
+                   }
+
 DOMAIN = 'rpi_gpio'
 
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_BOARD_FAMILY, default=DEFAULT_FAMILY): cv.string,
+        vol.Optional(CONF_BOARD): cv.string
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+GPIO_LIBRARY = None
+
+
+class UnknownBoardFamily(Exception):
+    """board_family should be 'raspberry_pi' or 'orange_pi'"""
+    pass
+
+
+class UnknownOrangePiBoard(Exception):
+    """'board' config item not set"""
+    pass
 
-def setup(hass, config):
-    """Set up the Raspberry PI GPIO component."""
-    from RPi import GPIO  # pylint: disable=import-error
+
+def setup(hass, base_config):
+    """Set up the GPIO component."""
+    global GPIO_LIBRARY
```

Comment:
Don't use `global`, we have `hass.data`.

Reply:
Ok, I'll go have a look at examples.

Reply:
Done.